homeassistant/components/unifiprotect/binary_sensor.py

```diff
@@ -219,24 +210,24 @@ def _async_update_device_from_protect(self) -> None:
             is_ringing = (
                 False if last_ring is None else (now - last_ring) < RING_INTERVAL
             )
+            _LOGGER.warning("%s, %s, %s", last_ring, now, is_ringing)
             if is_ringing:
                 if self._doorbell_callback is not None:
-                    self._doorbell_callback.cancel()
-                self._doorbell_callback = asyncio.ensure_future(
-                    self._async_wait_for_doorbell(last_ring + RING_INTERVAL)
+                    _LOGGER.debug("Canceling doorbell ring callback")
+                    self._doorbell_callback()
+                    self._doorbell_callback = None
+                self._doorbell_callback = self.hass.helpers.event.async_call_later(
```

Comment:
The newer practice is to import `async_call_later` and pass it hass instead of using the bound method since it works better with typing.